content/en/blog/2023/waypoint-proxy-made-simple/index.md

```diff
@@ -0,0 +1,221 @@
+- name: inbound-vip|9080|http|reviews.default.svc.cluster.local
+ validateClusters: false
+ virtualHosts:
+ - domains:
+   - '*'
+   name: inbound|http|9080
+   routes:
+   - decorator:
+       operation: reviews:9080/*
+     match:
+       prefix: /
+     metadata:
+       filterMetadata:
+         istio:
+           config: /apis/networking.istio.io/v1alpha3/namespaces/default/virtual-service/reviews
+     route:
+       maxGrpcTimeout: 0s
+       retryPolicy:
+         hostSelectionRetryMaxAttempts: "5"
+         numRetries: 2
+         retriableStatusCodes:
+         - 503
+         retryHostPredicate:
+         - name: envoy.retry_host_predicates.previous_hosts
+           typedConfig:
+             '@type': type.googleapis.com/envoy.extensions.retry.host.previous_hosts.v3.PreviousHostsPredicate
+         retryOn: connect-failure,refused-stream,unavailable,cancelled,retriable-status-codes
+       timeout: 0s
+       weightedClusters:
+         clusters:
+         - name: inbound-vip|9080|http/v1|reviews.default.svc.cluster.local
+           weight: 90
+         - name: inbound-vip|9080|http/v2|reviews.default.svc.cluster.local
+           weight: 10
+{{< /text >}}
+
+Check out the endpoints for `reviews` waypoint proxy:
+
+{{< text bash >}}
+$ istioctl proxy-config endpoints deploy/bookinfo-reviews-istio-waypoint
+ENDPOINT                                            STATUS  OUTLIER CHECK CLUSTER
+127.0.0.1:15000                                     HEALTHY OK            prometheus_stats
+127.0.0.1:15020                                     HEALTHY OK            agent
+envoy://connect_originate/                          HEALTHY OK            encap
+envoy://connect_originate/10.244.1.6:9080           HEALTHY OK            inbound-vip|9080|http/v2|reviews.default.svc.cluster.local
+envoy://connect_originate/10.244.1.6:9080           HEALTHY OK            inbound-vip|9080|http|reviews.default.svc.cluster.local
+envoy://connect_originate/10.244.2.11:9080          HEALTHY OK            inbound-vip|9080|http/v1|reviews.default.svc.cluster.local
+envoy://connect_originate/10.244.2.11:9080          HEALTHY OK            inbound-vip|9080|http|reviews.default.svc.cluster.local
+envoy://connect_originate/10.244.2.14:9080          HEALTHY OK            inbound-vip|9080|http/v3|reviews.default.svc.cluster.local
+envoy://connect_originate/10.244.2.14:9080          HEALTHY OK            inbound-vip|9080|http|reviews.default.svc.cluster.local
+envoy://main_internal/                              HEALTHY OK            main_internal
+unix://./etc/istio/proxy/XDS                        HEALTHY OK            xds-grpc
+unix://./var/run/secrets/workload-spiffe-uds/socket HEALTHY OK            sds-grpc
+{{< /text >}}
+
+Note you don’t get any endpoints related to any other services other than reviews, even though you have a few other services in the `default` and `istio-system` namespace.
```

Comment:
I feel like we're missing a final step where the waypoint routes the request to reviews by encap'ing in HBONE.

Reply:
Yep, i didn't mention it because it is not directly related to the blog topic (which is waypoint config made simpler) - there is nothing for me to highlight as part of trimmed config for encap hbone.